src/firefly/js/charts/ui/PinnedChartPanel.jsx

```diff
@@ -139,7 +140,30 @@ export const PinChart = ({viewerId, tbl_group}) => {
     return canPin ? <TextButton onClick={doPinChart} title='Pin the active chart'>Pin Chart</TextButton> : null;
 };
 
-export function pinChart({chartId, autoLayout=true}) {
+export function pinChart({chartId, autoLayout=true }) {
+
+    // there are cases when chart is not fully loaded.  if so, wait before pinning the chart
+    if (isChartLoaded(chartId)) {
+        doPinChart({chartId,autoLayout});
+    } else {
+        dispatchAddActionWatcher({actions:[CHART_UPDATE], callback: (action, cancelSelf) => {
+                const aChartId = action.payload?.chartId;
+                if (chartId === aChartId) {
+                    if (isChartLoaded(chartId)) {
+                        doPinChart({chartId, autoLayout});
+                        cancelSelf && cancelSelf();
```

Comment:
It's from my `copy/paste`.  Can you change `cancelSelf && cancelSelf()` to `cancel?.()`?